src/blister.c

```diff
@@ -1315,11 +1296,10 @@ Obj FuncUNITE_BLIST_LIST (
     lenList  = LEN_LIST( list );
 
     RequireBlist("UniteBlistList", blist, "blist");
-    while ( LEN_BLIST(blist) != lenList ) {
-        blist = ErrorReturnObj(
-          "UniteBlistList: <blist> must have the same length as <list> (%d)",
-            lenList, 0L,
-            "you can replace <blist> via 'return <blist>;'" );
+    if (LEN_BLIST(blist) != lenList) {
+        ErrorMayQuit("UniteBlistList: <blist> must have the same length as "
+                     "<list> (%d)",
+                     lenList, 0L);
```

Comment:
Perhaps use `CheckSameLength` here, too? The overhead of computing the list length again should be negligible, shouldn't it?